var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -0,0 +1,497 @@
+
+        cfg.write("#------------------{0}\n".format("-" * 60))
+        cfg.write("# MPI\n")
+        cfg.write("#------------------{0}\n\n".format("-" * 60))
+
+        if "+mpi" in spec:
+            cfg.write(cmake_cache_option("ENABLE_MPI", True))
+            cfg.write(cmake_cache_entry("MPI_C_COMPILER", spec['mpi'].mpicc))
+            cfg.write(cmake_cache_entry("MPI_CXX_COMPILER",
+                                        spec['mpi'].mpicxx))
+            if "+fortran" in spec or f_compiler is not None:
+                cfg.write(cmake_cache_entry("MPI_Fortran_COMPILER",
+                                            spec['mpi'].mpifc))
+
+            # Determine MPIEXEC
+            if on_blueos:
+                mpiexec = join_path(spec['mpi'].prefix.bin, 'mpirun')
+            else:
+                mpiexec = join_path(spec['mpi'].prefix.bin, 'mpiexec')
+                if not os.path.isfile(mpiexec):
+                    mpiexec = "/usr/bin/srun"
+            # starting with cmake 3.10, FindMPI expects MPIEXEC_EXECUTABLE
+            # vs the older versions which expect MPIEXEC
+            if self.spec["cmake"].satisfies('@3.10:'):
+                cfg.write(cmake_cache_entry("MPIEXEC_EXECUTABLE", mpiexec))
+            else:
+                cfg.write(cmake_cache_entry("MPIEXEC", mpiexec))
+
+            # Determine MPIEXEC_NUMPROC_FLAG
+            if on_blueos:
+                cfg.write(cmake_cache_entry("MPIEXEC_NUMPROC_FLAG", "-np"))
+                cfg.write(cmake_cache_entry("BLT_MPI_COMMAND_APPEND",
+                                            "mpibind"))
+            else:
+                cfg.write(cmake_cache_entry("MPIEXEC_NUMPROC_FLAG", "-n"))
+        else:
+            cfg.write(cmake_cache_option("ENABLE_MPI", False))
+
+        ##################################
+        # Devtools
+        ##################################
+
+        cfg.write("#------------------{0}\n".format("-" * 60))
+        cfg.write("# Devtools\n")
+        cfg.write("#------------------{0}\n\n".format("-" * 60))
+
+        # Add common prefix to path replacement list
+        if "+devtools" in spec:
+            # Grab common devtools root and strip the trailing slash
+            path1 = os.path.realpath(spec["uncrustify"].prefix)
+            path2 = os.path.realpath(spec["doxygen"].prefix)
+            devtools_root = os.path.commonprefix([path1, path2])[:-1]
+            path_replacements[devtools_root] = "${DEVTOOLS_ROOT}"
+            cfg.write("# Root directory for generated developer tools\n")
+            cfg.write(cmake_cache_entry("DEVTOOLS_ROOT", devtools_root))
+
+        if "python" in spec or "devtools" in spec:
+            python_bin_dir = get_spec_path(spec, "python",
+                                           path_replacements, use_bin=True)
+            cfg.write(cmake_cache_entry("PYTHON_EXECUTABLE",
+                                        pjoin(python_bin_dir, "python")))
```

Comment:
This is only true for Python 2. For Python 3, the `python` executable may or may not exist. You should instead use `spec['python'].command.path`